### DISCUSSION TITLE
Notifier Trace logs: use of constant NOTIFIER_TRACE

### DISCUSSION THREAD

**Discussion by torvista** [zencart/zencart#5786] [Ideas]
While chasing my tail I discover that there is an apparently undocumented feature.
If this constant is defined
https://github.com/zencart/zencart/blob/23066d01cc9b5c6922eebfa206a087ad3c8e5bbb/includes/classes/traits/NotifierManager.php#L99
a notifier_trace log is created.
If present in for example, includes/extra_configures: define('NOTIFIER_TRACE', 'true')
whether storefont and/or admin, it produces a log for either or both, appending to the same file.

While considering the SQL to add this to the admin->Logging options, I think it should be further modified to provide for independently logging admin and storefront as per the original code: 
https://github.com/lat9/zen_trace

And, eventually to be documented.

**Comment by lat9** [comment]
... and to:
1. Add a date/time portion to the log's name.  Otherwise, that log gets really big really fast.
2. Possibly (?) use the `EXCLUDE_ADMIN_IP_FOR_MAINTENANCE` setting to identify IP addresses for which the notification-logs should be created.  Otherwise, it's quite the mess trying to use this on a live site.
